### PR TITLE
Add net/http.Transport.RoundTrip()

### PR DESCRIPTION
**Pull request by sn00pyd0g3**
Add the function #504

### CODE REVIEW

**Review by sauyon — CHANGES_REQUESTED** [review]
I'd prefer you just added `this.getTarget().(Method).hasQualifiedName("net/http", "Transport", "RoundTrip")` and / or make a new class for it.

**Comment by sn00pyd0g3** [comment]
add a new class, it's very similar to ClientDo

File: ql/src/semmle/go/frameworks/stdlib/NetHttp.qll [hunk 1]
```diff
@@ -216,6 +219,35 @@ module NetHttp {
     }
   }
 
+  /** A call to the Transport.RoundTrip function in the `net/http` package. */
```
Comment (intrigus-lgtm) [inline ql/src/semmle/go/frameworks/stdlib/NetHttp.qll:222]:
```suggestion
  /** A call to the `Transport.RoundTrip` function in the `net/http` package. */
```

**Comment by smowton** [comment]
@sn00pyd0g3 please rebase rather than introduce merges from our `main`

**Pull request closed without merging**